analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsShouldBeLogged.cs

```diff
@@ -232,12 +232,30 @@ public void PartialLogging()
         try { }
         catch (Exception e)
         {
-            logger.LogWarning(new EventId(1), e.InnerException, "Message!");        // Noncompliant
+            logger.LogWarning(new EventId(1), e.StackTrace, "Message!");        // Noncompliant
             logger.LogWarning(new EventId(1), (e.Message != null).ToString());
 //          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Secondary
         }
     }
 
+    public void LogInnerException()
```

Comment:
Some corner cases to covered (I wouldn't bother adjusting the implementation, just to document it):

```cs
logger.LogWarning(new EventId(1), e?.InnerException, "Message!");  // conditional access
logger.LogWarning(new EventId(1), e.InnerException.InnerException, "Message!"); // nested InnerException
logger.LogWarning(new EventId(1), (Exception)e.InnerException, "Message!"); // conversion
```